### DISCUSSION TITLE
请问如何配置vscode的layaair源码调试环境？

### DISCUSSION THREAD

**Discussion by yangshuang-developer** [layabox/LayaAir#1361] [General]
1，2.9的源码，适配的node版本是多少？这个尝试了好多个都有问题。
2，要如何才能debug layaair引擎的源码，想单步调试跟一下底层逻辑
3，修改的引擎，要build之后放到工程，然后才能验证是否改好了，有办法直接用引擎源码的来跑工程吗？

**Comment by Lurenyiabc** [comment]
1.我们常用的node版本是10.15.1
2.目前调试还是依赖的谷歌
3.修改完之后可以尝试跑一跑相关的引擎示例，直接跑工程可以试试ide的源码版本的ts（不太推荐，最好有过ts编译相关的经验）

**Comment by yangshuang-developer** [comment]
> 1.我们常用的node版本是10.15.1
> 2.目前调试还是依赖的谷歌
> 3.修改完之后可以尝试跑一跑相关的引擎示例，直接跑工程可以试试ide的源码版本的ts（不太推荐，最好有过ts编译相关的经验）

感谢回答，我试试node10.15.1，尝试过的12.xx，13.xx,14.xx版本都又各种问题，gulp和rollup都不太兼容
2，请问下这个怎么开启debug，我修改了tscoifig里面的sourceMap，然后修改了lauch.json指向indexTSC.html，但是无法调试，无法打开，各种奇怪的报错。 如果开启了服务器，从网页中访问可以打开，但是又无法调试，可以有文档指引下如何使用源码吗？主页上挂的那个文档，操作性太差。

**Comment by yangshuang-developer** [comment]
> 修改完之后可以尝试跑一跑相关的引擎示例

请问下这个要怎么配置debug的环境

**Comment by lirongf** [comment]
一般是直接开服务器  用浏览器来调试的

**Comment by lirongf** [comment]
https://ldc2.layabox.com/doc/?nav=zh-ts-1-0-3
你可以看看这篇文档

**Comment by yangshuang-developer** [comment]
> https://ldc2.layabox.com/doc/?nav=zh-ts-1-0-3
> 你可以看看这篇文档

这个看过好多遍了

**Comment by yangshuang-developer** [comment]
> 一般是直接开服务器 用浏览器来调试的

这个调试是说再f12中断点了看吗？之前一直是在vscode中下的断点，无法触发debug

**Comment by jinghailang** [comment]
/usr/bin/env 'NODE_OPTIONS=--require "/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/ms-vscode.js-debug/src/bootloader.bundle.js" --inspect-publish-uid=http' 'VSCODE_INSPECTOR_OPTIONS={"inspectorIpc":"/var/folders/m6/jxpc4_710yxgtbymplyn615c0000gn/T/node-cdp.72491-1.sock","deferredMode":false,"waitForDebugger":"","execPath":"/usr/local/bin/node","onlyEntrypoint":false,"autoAttachMode":"always","fileCallback":"/var/folders/m6/jxpc4_710yxgtbymplyn615c0000gn/T/node-debug-callback-34a264824215f26b"}' /usr/local/bin/node ./node_modules/gulp/bin/gulp.js LayaAirBuild 
Debugger attached.
[15:40:42] Using gulpfile ~/Desktop/LayaAir/gulpfile.js
[15:40:42] Starting 'LayaAirBuild'...
[15:40:42] Starting 'tsc'...
Error: spawn /Users/admin/Desktop/LayaAir/node_modules/.bin/tsc.cmd ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
    at onErrorNT (internal/child_process.js:465:16)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn /Users/admin/Desktop/LayaAir/node_modules/.bin/tsc.cmd',
  path: '/Users/admin/Desktop/LayaAir/node_modules/.bin/tsc.cmd',
  spawnargs: [ '-b', 'src/samples/tsconfig.json' ],
  cmd: '/Users/admin/Desktop/LayaAir/node_modules/.bin/tsc.cmd -b src/samples/tsconfig.json'
} 
  
 
[15:40:42] Finished 'tsc' after 15 ms
[15:40:42] Starting 'LayaAirShaderCopy'...
[15:40:42] Finished 'LayaAirShaderCopy' after 706 ms
[15:40:42] Finished 'LayaAirBuild' after 724 ms
Waiting for the debugger to disconnect...


VSCode ctr+F5 编译报错，mac系统，按照https://ldc2.layabox.com/doc/?nav=zh-ts-1-0-3文档操作的，anywhere启动web服务也不能成功
